File: java_implementation/src/main/java/com/tambapps/web/page_scrapping/parameters/validator/ExistingDirectory.java
package com.tambapps.web.page_scrapping.parameters.validator;

import com.beust.jcommander.IValueValidator;
import com.beust.jcommander.ParameterException;

import java.io.File;

public class ExistingDirectory implements IValueValidator<File> {

  @Override
  public void validate(String name, File value) throws ParameterException {
    if (!value.exists()) {
      throw new ParameterException("Directory with path" + value.getPath() + " doesn't exists");
    }
    if (!value.isDirectory()) {
      throw new ParameterException("File with path " + value.getPath() + " isn't a directory");
    }
  }

}

File: kotlin_implementation/src/main/kotlin/com/tambapps/kotlin/parameters/validator/ExistingDirectory.kt
package com.tambapps.kotlin.parameters.validator

import com.beust.jcommander.IValueValidator
import com.beust.jcommander.ParameterException
import java.io.File

class ExistingDirectory: IValueValidator<File> {
    override fun validate(name: String, value: File) {
        if (!value.exists()) {
            throw ParameterException("Directory with path ${value.path} doesn't exists")
        }
        if (!value.isDirectory) {
            throw ParameterException("File with path ${value.path} isn't a directory")
        }
    }
}